### DISCUSSION TITLE
[feature] Allow configuring globally ignored files in `sgconfig.yml`

### DISCUSSION THREAD

**Discussion by soceanainn** [ast-grep/ast-grep#1495] [General]
# ⭐ Suggestion

Add `ignore` option to `sgconfig.yml` that would allow specifying a list of files / paths to ignore.
 
# 💻 Use Cases

I have some files that are checked in to VCS that are autogenerated, and I don't want to have ast-grep applied to them.

Maintaining and syncing a set of ignores across every rule  (which may or may not be extended by each individual rule) is troublesome.

For now - creating a `.ignore` file specifically for the purposes of managing ignores in `ast-grep` works, but it's a bit of an annoyance as it means we now need multiple configuration files just to manage ast-grep, which is just a single tool out of a large number of tools used in this project.

**Comment by HerringtonDarkholme** [comment]
Thanks for using ast-grep!

This is indeed a good question. I will put it on the roadmap and think about how the configuration should be and also how it should interact with .ignore and other files.

**Comment by HerringtonDarkholme** [comment]
After some investigation, I fount that ignoring files isn't a straightforward issue in ast-grep. Reason

* for other searching tools, they respect `.gitignore` and `.ignore`. ripgrep has one more `rgignore` but silversearch ag removes `agignore` https://github.com/ggreer/the_silver_searcher/wiki/Advanced-Usage#ignoring-files
* for other linting tools, they have ignore/exclude settings but it mostly overlaps with `.gitignore` or search ignoring. example: https://docs.astral.sh/ruff/settings/#exclude 
* ast-grep itself has two features: searching and lining. Should ignore apply to both `sg run` and `sg scan`? 
* should ast-grep provide options to distinguish run ignore and scan ignore?
* if users are using multiple tools, is it better to use `.ignore` to reduce configuration across different tools?

Given the issue above, I am inclining to keep the status quo and will put this feature in the backlog at the moment.

If you are interested in the feature and has your user story, please comment below and I will reconsider it if more users are requesting this.

**Reply by adamsol** [reply]
An option to ignore files via config file could be useful. For example, the `scan` command already has the [`--globs`](https://ast-grep.github.io/reference/cli/scan.html#globs-globs) option, but it's impossible to set in the config file. This means we may need to keep options split between two different places.

My use-case is linting a repository which contains some folders with external libraries that should not be linted. Tools like Ruff allow listing the excluded folders in the config file.
